go/pipeline/batch.go

```diff
@@ -0,0 +1,396 @@
+//	A new ClusterBatchRetryStrategy instance.
+func NewClusterBatchRetryStrategy() *ClusterBatchRetryStrategy {
+	return &ClusterBatchRetryStrategy{false, false}
+}
+
+// WithRetryServerError configures whether to retry on server errors.
+//
+// Parameters:
+//
+//	retryServerError - If true, retry on server errors.
+//
+// Returns:
+//
+//	The updated ClusterBatchRetryStrategy instance.
+func (cbrs *ClusterBatchRetryStrategy) WithRetryServerError(retryServerError bool) *ClusterBatchRetryStrategy {
+	cbrs.RetryServerError = retryServerError
+	return cbrs
+}
+
+// WithRetryConnectionError configures whether to retry on connection errors.
+//
+// Parameters:
+//
+//	retryConnectionError - If true, retry on connection errors.
+//
+// Returns:
+//
+//	The updated ClusterBatchRetryStrategy instance.
+func (cbrs *ClusterBatchRetryStrategy) WithRetryConnectionError(retryConnectionError bool) *ClusterBatchRetryStrategy {
+	cbrs.RetryConnectionError = retryConnectionError
+	return cbrs
+}
+
+// NewStandaloneBatchOptions creates a new options instance for standalone batches.
+//
+// Returns:
+//
+//	A new StandaloneBatchOptions instance.
+func NewStandaloneBatchOptions() *StandaloneBatchOptions {
+	return &StandaloneBatchOptions{}
+}
+
+// TODO support duration
+
+// WithTimeout sets the timeout for the batch execution.
+//
+// Parameters:
+//
+//	timeout - The timeout in milliseconds.
+//
+// Returns:
+//
+//	The updated StandaloneBatchOptions instance.
+func (sbo *StandaloneBatchOptions) WithTimeout(timeout uint32) *StandaloneBatchOptions {
+	sbo.Timeout = &timeout
+	return sbo
+}
+
+// NewClusterBatchOptions creates a new options instance for cluster batches.
+//
+// Returns:
+//
+//	A new ClusterBatchOptions instance.
+func NewClusterBatchOptions() *ClusterBatchOptions {
+	return &ClusterBatchOptions{}
+}
+
+// WithTimeout sets the timeout for the batch execution.
+//
+// Parameters:
+//
+//	timeout - The timeout in milliseconds.
+//
+// Returns:
+//
+//	The updated ClusterBatchOptions instance.
+func (cbo *ClusterBatchOptions) WithTimeout(timeout uint32) *ClusterBatchOptions {
+	cbo.Timeout = &timeout
+	return cbo
+}
+
+// TODO ensure only single node route is allowed (use config.NotMultiNode?)
+
+// WithRoute sets the routing strategy for the batch.
+//
+// Parameters:
+//
+//	route - The routing strategy to use.
+//
+// Returns:
+//
+//	The updated ClusterBatchOptions instance.
+func (cbo *ClusterBatchOptions) WithRoute(route config.Route) *ClusterBatchOptions {
+	cbo.Route = &route
+	return cbo
+}
+
+// WithRetryStrategy sets the retry strategy for the batch.
+//
+// Parameters:
+//
+//	retryStrategy - The retry strategy to use.
+//
+// Returns:
+//
+//	The updated ClusterBatchOptions instance.
+func (cbo *ClusterBatchOptions) WithRetryStrategy(retryStrategy ClusterBatchRetryStrategy) *ClusterBatchOptions {
+	cbo.RetryStrategy = &retryStrategy
+	return cbo
+}
+
+// ====================
+
+// TODO - move this struct and convert methods to internals
```

Comment:
If this is moved to internal, how would anyone use this Option? Why not just define this as `batchOptions` which would prevent creating the type, while still allowing the use of the type.